libcudacxx/include/cuda/std/__type_traits/type_list.h

```diff
@@ -557,6 +557,82 @@ using __type_front = __type_at_c<0, _List>;
 template <class _List>
 using __type_back = __type_at_c<_List::__size - 1, _List>;
 
+//! \brief A pair of types
+template <class _First, class _Second>
+struct _CCCL_TYPE_VISIBILITY_DEFAULT __type_pair
+{
+  using __first _CCCL_NODEBUG_ALIAS  = _First;
+  using __second _CCCL_NODEBUG_ALIAS = _Second;
+};
+
+//! \brief Retreive the first of a pair of types
+//! \pre \c _Pair is a specialization of \c __type_pair
+template <class _Pair>
+using __type_pair_first = typename _Pair::__first;
+
+//! \brief Retreive the second of a pair of types
+//! \pre \c _Pair is a specialization of \c __type_pair
+template <class _Pair>
+using __type_pair_second = typename _Pair::__second;
+
+//! \see __type_switch
+template <class _Value>
+using __type_default _CCCL_NODEBUG_ALIAS = __type_pair<void, _Value>;
+
+//! \see __type_switch
+template <class _IntType, _IntType _Label, class _Value>
+using __type_case _CCCL_NODEBUG_ALIAS = __type_pair<integral_constant<_IntType, _Label>, _Value>;
```

Comment:
Right now I only see us using `size_t` and `int` as choices. Could we simplify the whole thing to 

```cpp
template<size_t _Choice, class _Value>
struct __type_case;
```